tests/object_sync_test.go

```diff
@@ -1049,6 +1213,685 @@ func makeTestNotifications(t *testing.T) []Notification {
+	if len(g.Children) == 0 {
+		if g.IsRedundancyGroup() {
+			// Verify that the redundancy group was deleted from the database after removing all of its children.
+			// Requires that this redundancy group has been verified before uisng this method and its ID is set accordingly.
+			require.NotNilf(t, g.Id, "ID should be set for redundancy group %q", g.Name)
+
+			var exists bool
+			err := db.Get(&exists, db.Rebind(`SELECT EXISTS(SELECT 1 FROM redundancy_group WHERE id = ?)`), g.Id)
+			assert.NoErrorf(t, err, "fetching redundancy group %q with ID %q", g.Name, g.Id)
+			assert.Falsef(t, exists, "redundancy group %q with ID %q should not exist", g.Name, g.Id)
+
+			err = db.Get(&exists, db.Rebind(`SELECT EXISTS(SELECT 1 FROM redundancy_group_state WHERE id = ?)`), g.Id)
+			assert.NoErrorf(t, err, "fetching redundancy group state by ID %q", g.Id)
+			assert.Falsef(t, exists, "redundancy group state with ID %q should not exist", g.Id)
+		}
+
+		return
+	}
+
+	totalExpectedChildren := g.Children
+	totalExpectedParents := g.Parents
+	// Retrieve all children and parents of all the dependency groups of this subtest.
+	for _, sibling := range dependencyGroupTest.Groups {
+		for _, child := range sibling.Children {
+			if !slices.Contains(totalExpectedChildren, child) {
+				totalExpectedChildren = append(totalExpectedChildren, child)
+			}
+		}
+
+		for _, parent := range sibling.Parents {
+			if !slices.Contains(totalExpectedParents, parent) {
+				totalExpectedParents = append(totalExpectedParents, parent)
+			}
+		}
+	}
+
+	// Fetch the redundancy group referenced by the children hosts as their parent node (if any).
+	// If the dependency group is a redundancy group, the redundancy group itself should be the parent node
+	// and if the dependency serialization is correct, we should find only a single redundancy group.
+	query, args, err := sqlx.In(`SELECT rg.id, rg.display_name FROM redundancy_group rg
+		INNER JOIN dependency_node parent ON parent.redundancy_group_id = rg.id
+		INNER JOIN dependency_edge tn ON tn.to_node_id = parent.id
+		INNER JOIN dependency_node child ON child.id = tn.from_node_id
+		INNER JOIN host child_host ON child_host.id = child.host_id
+		INNER JOIN redundancy_group_state rgs ON rgs.redundancy_group_id = rg.id
+		WHERE child_host.name IN (?)`,
+		totalExpectedChildren,
+	)
+	require.NoError(t, err, "expanding SQL IN clause for redundancy groups query")
+
+	var redundancyGroups []DependencyGroup
+	err = db.Select(&redundancyGroups, db.Rebind(query+` AND rg.display_name = ? GROUP BY rg.id, rg.display_name`), append(args, g.Name)...)
+	require.NoError(t, err, "fetching redundancy groups")
+
+	var redundancyGroup DependencyGroup
+	if g.IsRedundancyGroup() {
+		require.Lenf(t, redundancyGroups, 1, "there should be exactly one redundancy group %q", g.Name)
+		redundancyGroup = redundancyGroups[0]
+		g.Id = redundancyGroup.Id
+		assert.Equal(t, g.Name, redundancyGroup.Name, "redundancy group name should match")
```

Comment:
I think it could make sense to just use a dedicated type for the query result here instead of `DependencyGroup`. At least here, the code would then say more explicitly that for a redundancy group, that group's name should be used as the display name compared to how this now hidden inside the `db` tags of the struct fields.